pkg/util/leaderlease/leaderlease.go

```diff
@@ -292,12 +296,14 @@ func (e *Etcd) waitExpiration(held bool, from uint64, stop chan struct{}) (bool,
 }
 
 // eventIndexFor returns the next etcd index to watch based on a response
-func eventIndexFor(resp *etcdclient.Response) uint64 {
-	if resp.Node != nil {
-		return resp.Node.ModifiedIndex + 1
-	}
-	if resp.PrevNode != nil {
-		return resp.PrevNode.ModifiedIndex + 1
+func eventIndexFor(resp *etcdclient.Response, useClusterLevelIndex bool) uint64 {
```

Comment:
If we are using the returned value to decide what to watch from, we should always use the cluster level index